lib/naughty/core.lua.in

```diff
@@ -237,6 +237,77 @@ function naughty.getById(id)
     end
 end
 
+--- Install expiration timer for notification object
+local function set_up_timer(notification, timeout)
+    local die = function (reason)
+        naughty.destroy(notification, reason)
+    end
+    if timeout > 0 then
+        local timer_die = timer { timeout = timeout }
+        timer_die:connect_signal("timeout", function() die(naughty.notificationClosedReason.expired) end)
+        if not suspended then
```

Comment:
Line 97 of this file. It is set by `naughty.suspend()` and `naughty.respond()` and it delays notifications until `respond` is called. (Think: "Cinema mode")